### PR TITLE
Add concurrent indexing/querying scenario to PMC workload

### PR DESCRIPTION
**Pull request by gkamat**
### Description
Add concurrent indexing/querying scenario to PMC workload.

### Testing
Tested both the standard scenario and the concurrent one against an OpenSearch cluster.

By submitting this pull request, I confirm that my contribution is made under the terms of the Creative Commons license.  Attribution: https://github.com/elastic/rally-tracks/pmc.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: pmc/test_procedures/default.json [hunk 1]
```diff
@@ -70,92 +71,50 @@
...
-          ,"target-throughput": 20
-          {%- elif target_throughput is string and target_throughput.lower() == 'none' %}
-          {%- else %}
-          ,"target-throughput": {{ target_throughput | tojson }}
-          {%- endif %}
-          {%- if search_clients is defined and search_clients %}
-          ,"clients": {{ search_clients | tojson}}
-          {%- endif %}
+          "iterations": 200,
+          "target-throughput": {{ target_throughput | default(20) | tojson }},
```
Comment (IanHoang) [inline pmc/test_procedures/default.json:75]: Much cleaner. Thanks!

File: pmc/test_procedures/indexing-querying.json [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  "name": "indexing-querying",
+  "description": "Indexes half the data for NYC Taxis, then concurrently queries OpenSearch while indexing again.",
+  "default": false,
+  "schedule": [
+    {
+      "operation": {
+        "operation-type": "put-settings",
+        "body": {
+          "transient": {
+            "search.default_search_timeout": "{{default_search_timeout | default(-1)}}"
```
Comment (IanHoang) [inline pmc/test_procedures/indexing-querying.json:11]: Are we planning to set this field to other workloads as well?

**Review by IanHoang — APPROVED** [review]

File: pmc/test_procedures/default.json (hunk 1 above)
Comment (gkamat) [reply pmc/test_procedures/default.json:75]: Yes, using "0" is the same as the erstwhile "none".

File: pmc/test_procedures/indexing-querying.json (hunk 2 above)
Comment (gkamat) [reply pmc/test_procedures/indexing-querying.json:11]: Yes, we'll add this one at a time.

**Pull request merged**